app/Http/Controllers/QuizController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreQuizResultRequest;
+use App\Models\Quiz;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class QuizController extends Controller
+{
+    public function show(Quiz $quiz): View
+    {
+        $questions = $quiz->questions;
+
+        return view('quizzes.show', [
+            'quiz' => $quiz,
+            'title' => $quiz->title,
+            'questions' => $questions,
+        ]);
+    }
+
+    public function storeQuizResult(StoreQuizResultRequest $request, Quiz $quiz): View
+    {
+        $data = $request->validated();
+        $user = Auth::user();
+
+        $numberOfQuestions = 0;
```

Comment:
You can get the count of the questions from the collection or relationship, so you don't need to manually increment a counter.

```suggestion
        $numberOfQuestions = $quiz->questions->count();
```

Reply:
My concern was that calling quiz->questions would call another query. @jobara and I discussed about this, and we could potentially store $quiz->questions to a variable, then get how many questions are there, or use [withCount](https://laravel.com/docs/9.x/eloquent-relationships#counting-related-models) to get the counts.